WordPress/gutenberg
# Migrate site editor feature preferences to interface package

## Description
Partly addresses #31965. Related to #31965, #34135 and #34154.

Migrates site editor feature preferences over to the interface package.

As they've never hit core, I've deprecated the `toggleFeature` action and `isFeatureActive` selector and will remove them from a future version.

The way editor settings work in the site editor is slightly different to other editors, I guess it has been augmented for the theme.json.

Typically in other editors the settings are merged with the feature preferences using `useMemo` in the 'Editor' component before being passed into the `EditorProvider`, but the site editor uses a memoized selector to combine the values. This wouldn't work with this refactor, because the values are now in a different store.

The options:
- Use a registry selector. I don't think would be good, because from what I understand registry selectors can't be memoized.
- Follow the approach of other editors and `useMemo` in the Editor component. This is what I've done in this PR, but it's open to feedback. The downside is that the `focusMode` and `fixedToolbar` properties are no longer available when using the selector.
- Migrate from the selector to a `useEditorSettings` hook that combines settings with preferences. This might be the way to go. There's some prior art in the site editor with a `useSetting` hook - https://github.com/WordPress/gutenberg/blob/914f543e1e50d8c65c05a417efbdf539ef0d7a63/packages/edit-site/src/components/editor/utils.js#L101-L112
- Something else. Maybe editor settings should be reconsidered as an immutable object and other mutable values passed to the editor provider separately.

## How has this been tested?
1. Checkout trunk and open the site editor.
2. Set some preferences to a value different to the default (e.g. 'Top toolbar' and 'Spotlight Mode' to active.)
3. Checkout this branch, rebuild and open up the site editor
4. The preferences should be retained from step 2.
5. Try toggling top toolbar and spotlight mode settings to ensure they work.

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
